codegen-lite/src/test/resources/software/amazon/awssdk/codegen/lite/regions/region-metadata-provider.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2013-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package software.amazon.awssdk.regions;
+
+import java.util.AbstractMap;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import software.amazon.awssdk.annotations.Generated;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.regions.regionmetadata.ApNortheast1;
+import software.amazon.awssdk.regions.regionmetadata.ApNortheast2;
+import software.amazon.awssdk.regions.regionmetadata.ApSouth1;
+import software.amazon.awssdk.regions.regionmetadata.ApSoutheast1;
+import software.amazon.awssdk.regions.regionmetadata.ApSoutheast2;
+import software.amazon.awssdk.regions.regionmetadata.CaCentral1;
+import software.amazon.awssdk.regions.regionmetadata.CnNorth1;
+import software.amazon.awssdk.regions.regionmetadata.CnNorthwest1;
+import software.amazon.awssdk.regions.regionmetadata.EuCentral1;
+import software.amazon.awssdk.regions.regionmetadata.EuWest1;
+import software.amazon.awssdk.regions.regionmetadata.EuWest2;
+import software.amazon.awssdk.regions.regionmetadata.EuWest3;
+import software.amazon.awssdk.regions.regionmetadata.SaEast1;
+import software.amazon.awssdk.regions.regionmetadata.UsEast1;
+import software.amazon.awssdk.regions.regionmetadata.UsEast2;
+import software.amazon.awssdk.regions.regionmetadata.UsGovWest1;
+import software.amazon.awssdk.regions.regionmetadata.UsWest1;
+import software.amazon.awssdk.regions.regionmetadata.UsWest2;
+
+@Generated("software.amazon.awssdk:codegen")
+@SdkPublicApi
+public final class RegionMetadataProvider {
```

Comment:
Should we keep this an interface if we decide to support runtime discovery later on?

Reply:
Makes sense. Will do.